### ISSUE TITLE
Unbound and DNS-over-QUIC (DoQ)

### ISSUE THREAD

**Issue by iz8mbw** [NLnetLabs/unbound#743]
Hello.
Any plan to implement DNS-over-QUIC (DoQ) in unbound for both forward upstream and downstream?

https://datatracker.ietf.org/doc/rfc9250/
https://datatracker.ietf.org/doc/html/rfc9250

Thanks!

**Comment by pettai** [comment]
https://github.com/NLnetLabs/unbound/tree/dnsoverquic
(I'm unsure about the current state...)

**Comment by wcawijngaards** [comment]
There is a pull request with the work in progress. It would be useful to know if it works. Meanwhile it is waiting for review and tests to be included. It can be found at #871 .

**Comment by pettai** [comment]
> There is a pull request with the work in progress. It would be useful to know if it works. Meanwhile it is waiting for review and tests to be included. It can be found at #871 .

Do you know if there is any software (other than `doqclient`) out there that can be used for testing?

**Comment by wcawijngaards** [comment]
No, I actually do not know of any. Nor do I know about servers running it.

**Comment by rail01** [comment]
@pettai [`dnslookup`](https://github.com/ameshkov/dnslookup) can be used to test DoQ queries, example QUIC-enabled DNS servers are `dns0.eu` and `dns.adguard.com`

**Comment by ghost** [comment]
Push

**Comment by Freso** [comment]
Looks like https://github.com/NLnetLabs/unbound/pull/871 was merged 9 months back and released with [1.22.0](https://github.com/NLnetLabs/unbound/releases/tag/release-1.22.0). Is something still missing before closing this issue?

**Comment by iz8mbw** [comment]
https://github.com/NLnetLabs/unbound/pull/871 implements doq for downstream, not also for upstream.
